### PR TITLE
Default Enarx trust process

### PR DESCRIPTION
**Pull request by axelsimon**
This PR is intended to replace enarx/rfcs#41, which through mistakes
on my part got messed up.

The vast majority of this RFC was written by @MikeCamel.

Tihs final draft does the following:
- incorporate all suggestions from the shared text editor document
  - changes by @lkatalin @ueno etc.
  - address discussions in design meeting 2020-06-15
- wrap everything at 80 characters
- reorganise "Requirements" section
- odds and ends

Signed-off-by: axelsimon <github@axelsimon.net>

### CODE REVIEW

File: 00004-default-enarx-trust-process/README.md [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+    - Enarx host agent
+    - Enarx runtime image
+    - other workloads
+ - tenant trust domain
+    - Orchestrator
+    - Enarx client agent
+    - tenant workload image
+    - tenant workload image repository
+
+#### State 1
```
Comment (lkatalin) [inline 00004-default-enarx-trust-process/README.md]: FYI this state is missing a label.

**Review by lkatalin — DISMISSED** [review]
LGTM, there's a missing label but I wouldn't block on this.

File: 00004-default-enarx-trust-process/README.md (hunk 1 above)
Comment (axelsimon) [reply 00004-default-enarx-trust-process/README.md]: FIxed, thanks!

**Comment by axelsimon** [comment]
This PR closes #5.

**Review by lkatalin — APPROVED** [review]

**Review by mbestavros — APPROVED** [review]

**Review by connorkuehl — APPROVED** [review]

**Pull request merged**